src/ray/core_worker/core_worker.cc

```diff
@@ -2048,6 +2048,18 @@ Status CoreWorker::CancelTask(const ObjectID &object_id,
   return Status::OK();
```

Comment:
Just confirming that the call above `return direct_task_submitter_->CancelTask(task_spec.value(), force_kill, recursive);` isn't blocking. For example if a user sent a list of `ObjectRef`s then if the above call is blocking then the tasks will be cancelled one after the other. If its non-blocking then the requests for cancelling will be sent and the above return statement will execute. Please let me know what's the case. Thanks.

Reply:
Well, the call is non-blocking because from the logs I see the function was called 3 times at exactly the same time instant. That means first all the requests are sent without waiting for their replies. This is cool. 

```
[2022-05-13 07:11:27,717 I 10768 7412] direct_task_transport.cc:599: Cancelling a task: c8ef45ccd0112571ffffffffffffffffffffffff01000000 force_kill: 0 recursive: 1
[2022-05-13 07:11:27,717 I 10768 7412] direct_task_transport.cc:599: Cancelling a task: 16310a0f0a45af5cffffffffffffffffffffffff01000000 force_kill: 0 recursive: 1
[2022-05-13 07:11:27,717 I 10768 7412] direct_task_transport.cc:599: Cancelling a task: c2668a65bda616c1ffffffffffffffffffffffff01000000 force_kill: 0 recursive: 1
```